include/OpenSpaceToolkit/Core/Types/String.hpp

```diff
@@ -18,11 +18,20 @@ namespace ostk
 {
 namespace core
 {
+
+// Forward declare Array to avoid circular dependency

```

Comment:
While I don't _love_ this, I think it is the recommended way to deal with circular dependencies.

Reply:
Yeah I think this is a proper case of using forward declarations.